### PR TITLE
🏃update e2e container to latest cluster-api

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:
we are still running with older capi code and repo, this update to latest 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: api/v1alpha3/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,4 +1,3 @@
-//go:build !ignore_autogenerated
```
Comment (chrischdi) [inline api/v1alpha3/zz_generated.deepcopy.go:1]: looks like our controller-gen in ci still adds these :-)

Comment (jichenjc) [reply api/v1alpha3/zz_generated.deepcopy.go:1]: I don't know what happened .. need further analysis
it should not report such error as I thought it's fixed already

**Comment by hidekazuna** [comment]
Can't we update go.mod as well?

**Comment by jichenjc** [comment]
you mean update to cluster-api 0.4.3 ? I can give a try

**Comment by hidekazuna** [comment]
> 
> 
> you mean update to cluster-api 0.4.3 ? I can give a try

Sure, update please.

**Review by chrischdi — COMMENTED** [review]
/approve

**Pull request merged**